process/block/shardblock.go

```diff
@@ -489,16 +489,19 @@ func (sp *shardProcessor) CommitBlock(
 		log.Debug(errNotCritical.Error())
 	}
 
-	processedMetaHdrs, errNotCritical := sp.getProcessedMetaBlocksFromPool(body)
+	processedMetaHdrs, errNotCritical := sp.getProcessedMetaBlocksFromPool(body, header)
 	if errNotCritical != nil {
 		log.Debug(errNotCritical.Error())
 	}
 
-	err = sp.saveLastNotarizedHeader(sharding.MetachainShardId, processedMetaHdrs)
-	if err != nil {
-		return err
+	errNotCritical = sp.saveLastNotarizedHeader(sharding.MetachainShardId, processedMetaHdrs)
```

Comment:
It is better to keep this critical. saveLastNotarizedHeader could return an error could not processed if lastNoterized.Nonce != maxNonce from processedMetaHdrs.

Reply:
Changed